daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions.tdml

```diff
@@ -2012,7 +2012,10 @@
       <tdml:documentPart type="text">abcd</tdml:documentPart>
     </tdml:document>
```

Comment:
Add a test that uses an expression like xs:string(../../foo/bar) where the expression is reused in contexts where ../../foo/bar have different contexts, and types. Since the path is just being passed to xs:string, this is polymorphic; however, if the expression were not being passed to xs:string but was just being reused to get a string, for example, and in one context ../../foo/bar is an int, the other a string, then we should get diagnostics that indicate the expression isn't correct in one or other other context, and the diagnostic should indicate both the expression location in the schema, AND the location that is problematic context.